lib/gateway/index.js

```diff
@@ -95,6 +99,9 @@ function bootstrap ({plugins, config} = {}) {
     rootRouter(req, res, next);
   });
 
+  app.use(notFoundHandler);
```

Comment:
do I understand correctly that it will always return 404  ignoring any error ? 
because it is just
```
function notFoundHandler (req, res) {
   res.egError(404);
 }
```

Reply:
No, notFoundHandler middleware will only be called when there is no error. To catch error you need to create middleware with `err, req, res, next` attributes.

Reply:
ah, yep, thx